### PR TITLE
[ci] Enforce a minimum Kotlin version in examples

### PR DESCRIPTION
**Pull request by stuartmorgan**
https://github.com/flutter/packages/pull/3973 caused an out-of-band failure after publishing, because an example that uses `url_launcher` had a too-old Kotlin version set. This is not something we consider client-breaking because `flutter` prodives a very clear error message with a straightforward and actionable fix step (update the app's Kotlin version), so the fix for the breakage is just to update our own examples.

Since increasingly we're likely to hit problems where modern version of dependencies don't work with old version of Kotlin, this adds repo-wide CI enforcement that examples are set to a minimum version (matching the current `flutter/flutter` template; we can increase this over time as we feel it's useful to do so).

### CODE REVIEW

**Comment by stuartmorgan** [comment]
CODEOWNERs forced a bunch of reviewers on this; @reidbaker is the only one who needs to actually review, since he should weigh in on this as something we enforce.

**Review by domesticmouse — APPROVED** [review]
To keep codeowners happy

**Review by reidbaker — APPROVED** [review]

File: script/tool/test/gradle_check_command_test.dart
Comment (reidbaker) [reply script/tool/test/gradle_check_command_test.dart:741]: I agree I think both should use a modified `minKotlinVersion` that is `@visibleForTesting`

Comment (stuartmorgan) [reply script/tool/test/gradle_check_command_test.dart]: Good point; done.

**Comment by stuartmorgan** [comment]
Landing on red to fix the tree.

**Pull request merged**
